source/codegen/templates/service.cpp.mako

```diff
@@ -45,6 +45,17 @@ namespace ${config["namespace_component"]}_grpc {
     }
   }
 % endif
+% if common_helpers.has_enum_array_string_out_param(functions):
+  template <typename T1, typename T2>
+  void ${service_class_prefix}Service::Copy(const T1& input, T2* output)
+  {
+    int i = 0;
+    for(auto item : input){
+      output->Add(item);
+      i++;
+    }
+  }
+% endif
```

Comment:
Add blank line before `endif` here and also after line 46 above, so that generated file has blank line after the function definition.

Reply:
added